File: backend/api/search/[query].js
const yahooFinanceService = require("../../services/yahooFinance");

// GET /api/search/:query
module.exports = async (req, res) => {
  try {
    const { query } = req.query; // dynamic segment
    if (!query || query.length < 1) {
      return res.status(400).json({ error: "Search query is required" });
    }

    const results = await yahooFinanceService.searchSymbols(query);
    res.json({ success: true, data: results });
  } catch (error) {
    console.error("Error in /api/search/[query]", error);
    res.status(500).json({ error: "Failed to search symbols", message: error.message });
  }
};

File: backend/api/search/[query].py
import yfinance as yf


def handler(req, res):
    # CORS headers
    res.set_header("Access-Control-Allow-Origin", "*")
    res.set_header("Access-Control-Allow-Methods", "GET,OPTIONS")
    res.set_header("Access-Control-Allow-Headers", "Content-Type, Authorization")
    if req.method == "OPTIONS":
        return res.status(200).send("")

    query = req.query.get("query")
    if not query:
        return res.status(400).json({"error": "Query is required"})

    try:
        # yfinance has limited search; attempt to get info for the provided symbol
        t = yf.Ticker(query.upper())
        info = t.info
        if "symbol" in info:
            results = [{
                "symbol": info.get("symbol", query.upper()),
                "name": info.get("longName", info.get("shortName", query.upper())),
                "exchange": info.get("exchange", "N/A"),
                "type": "EQUITY",
            }]
            return res.status(200).json({"success": True, "data": results})
        return res.status(200).json({"success": True, "data": []})
    except Exception:
        # Return empty results on error for search
        return res.status(200).json({"success": True, "data": []})


